### PR TITLE
Add architecture diagram, update docs

### PR DESCRIPTION
**Pull request by Patrik-Stas**
Signed-off-by: Patrik Stas <patrik.stas@absa.africa>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,6 +34,11 @@ However before you'll be able to pick one of these demos and run them locally, y
 all these demos depends on.  
 - [Building AriesVCX on OSX, Linux](./docs/build-general.md)
 
+# Project architecture
+The architecture is evolving - you can compare the diagram bellow with diagram under [roadmap](./roadmap.md).
```
Comment (mirgee) [inline README.md:38]: "below"

File: roadmap.md [hunk 2]
```diff
@@ -28,12 +28,11 @@ the library into 3 pieces.
    on Android, iOS and any programming language.
 
 # Phase 2 - In progress 🚧
+- 🚧 Migration from `Hyperledger libindy` to its successor [vdr-tools](https://gitlab.com/evernym/verity/vdr-tools).
```
Comment (mirgee) [inline roadmap.md:31]: Although most will probably understand what you mean, `Hyperledger libindy` is not stricly correct nomenclature and slightly confusing - I think `libindy` is accurate enough.  Whether and in what sense it is a successor is debatable (it is not a successor for some other projects), I would rather avoid the term "successor".

File: roadmap.md [hunk 3]
```diff
@@ -43,33 +42,12 @@ the library into 3 pieces.
...
-   version, as it contains asynchronous implementation of the storage module.
-   
-Migration from `libindy` to `vdr-tools` will likely be effortless relative to migration to `aries-askar`,
-so we will likely go for the second option.
-
-
-
+AriesVCX currently only supports single tenant architectures. In order to build scalable 
+institutional agents, we need to enable using in multi-tenant context - a single process 
+using `aries-vcx` should be able to manage multiple wallets/agents.
```
Comment (mirgee) [inline roadmap.md:47]: "AriesVCX currently does not support multitenancy - a single `aries-vcx` managing multiple wallets / agents. This feature is on our roadmap, however."

I don't believe in using 10 words where 2 would do :)

**Review by mirgee — CHANGES_REQUESTED** [review]

File: roadmap.md [hunk 4]
```diff
@@ -43,33 +42,10 @@ the library into 3 pieces.
...
-  [aries-credx](https://github.com/sovrin-foundation/aries-credx-framework-rs).
-2. Swap the `indy-sdk` version of `libindy` library for its [vdr-tools](https://gitlab.com/evernym/verity/vdr-tools)
-   version, as it contains asynchronous implementation of the storage module.
-   
-Migration from `libindy` to `vdr-tools` will likely be effortless relative to migration to `aries-askar`,
-so we will likely go for the second option.
-
-
-
+"AriesVCX currently can not manage more than 1 wallet simultaneously. Adding this capability is on our roadmap."
```
Comment (mirgee) [inline roadmap.md:45]: Please don't forget to remove the quotes :)

Comment (Patrik-Stas) [reply roadmap.md:45]: dannng :-D

**Review by mirgee — APPROVED** [review]

**Pull request merged**